qrscanner/src/main/java/io/gnosis/safe/qrscanner/QRCodeScanActivity.kt

```diff
@@ -46,11 +49,22 @@ class QRCodeScanActivity : AppCompatActivity() {
     }
 
     private fun finishWithResult(value: String) {
-        val result = Intent().apply { putExtra(RESULT_EXTRA, value) }
-        setResult(Activity.RESULT_OK, result)
-        finish()
+        if (isEthereumAddress(value)) {
+            val result = Intent().apply { putExtra(RESULT_EXTRA, value) }
+            setResult(Activity.RESULT_OK, result)
+            finish()
+        } else {
+            AlertDialog.Builder(this)
```

Comment:
can we just show snackbar and continue to capture screen automatically after short delay?